### PR TITLE
Add core module

### PR DESCRIPTION
**Pull request by acidmoose**
- Add entry point class for secondbase project (SecondBase)
- Add dynamic loading of Flags specific to submodules

### CODE REVIEW

File: core/src/main/java/org/secondbase/core/ModuleConfigManager.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * Generic management of submodules
+ */
+public class ModuleConfigManager {
+    // ServiceLoaded for classes that has configuration Flags
+    private final ServiceLoader<ConfigurableModule> configurableModule
+            = ServiceLoader.load(ConfigurableModule.class);
+
+    /**
+     * Load and configure modules
```
Comment (neumayer) [inline core/src/main/java/org/secondbase/core/ModuleConfigManager.java]: First comment line end with a dot?

File: core/src/main/java/org/secondbase/core/ModuleConfigManager.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package org.secondbase.core;
+
+import java.util.ServiceLoader;
+import org.secondbase.core.config.ConfigurableModule;
+import org.secondbase.flags.Flags;
+
+/**
+ * Generic management of submodules
```
Comment (neumayer) [inline core/src/main/java/org/secondbase/core/ModuleConfigManager.java]: Here too?

File: core/src/main/java/org/secondbase/core/ModuleConfigManager.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    // ServiceLoaded for classes that has configuration Flags
+    private final ServiceLoader<ConfigurableModule> configurableModule
+            = ServiceLoader.load(ConfigurableModule.class);
+
+    /**
+     * Load and configure modules
+     * @param flags Flags instance to load Flagged variables into
+     */
+    public ModuleConfigManager(final Flags flags) {
+        // Load Flags from all modules that has them
```
Comment (neumayer) [inline core/src/main/java/org/secondbase/core/ModuleConfigManager.java]: has->have

File: core/src/main/java/org/secondbase/core/SecondBase.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Coordination class for SecondBase modules. Handles Flags parsing and cooperation between modules.
+ */
+public class SecondBase {
+
+    private static final Logger LOG = Logger.getLogger(SecondBase.class.getName());
+    private Flags flags;
+
+    /**
```
Comment (neumayer) [inline core/src/main/java/org/secondbase/core/SecondBase.java:14]: Make a more understandable comment? Maybe "Initialise SecondBase object with default values"?

File: core/src/main/java/org/secondbase/core/SecondBase.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+     * Convenience constructor for:
+     *     new SecondBase(args, new Flags());
+     * @param args command line arguments
+     */
+    public SecondBase(final String[] args) {
+        init(args, new Flags());
+    }
+
+    /**
+     * Setup SecondBase
```
Comment (neumayer) [inline core/src/main/java/org/secondbase/core/SecondBase.java]: Setup -> Set up
Add dot here too?

File: core/src/main/java/org/secondbase/core/SecondBase.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    private void init(final String[] args, final Flags flags) {
+        this.flags = flags;
+
+        new ModuleConfigManager(flags);
+
+        flags.parse(args);
+
+        if (flags.helpFlagged()) {
+            flags.printHelp(System.out);
+            System.exit(0);
```
Comment (neumayer) [inline core/src/main/java/org/secondbase/core/SecondBase.java:40]: This is a postive exit case?

File: core/src/main/java/org/secondbase/core/config/ConfigurableModule.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package org.secondbase.core.config;
+
+import org.secondbase.flags.Flags;
+
+/**
+ * Implemented by modules that has configurable Flags
```
Comment (neumayer) [inline core/src/main/java/org/secondbase/core/config/ConfigurableModule.java]: Add dot. Make it a complete sentence?

File: core/src/main/java/org/secondbase/core/config/ConfigurableModule.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+ */
+public interface ConfigurableModule {
+    /**
+     * Load options for a ConfigurableModule
```
Comment (neumayer) [inline core/src/main/java/org/secondbase/core/config/ConfigurableModule.java]: Dot here too.

File: example/src/main/java/org/secondbase/example/main/HelloSecondBase.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
+package org.secondbase.example.main;
+
+import org.secondbase.core.SecondBase;
+import org.secondbase.flags.Flag;
+import org.secondbase.flags.Flags;
+
+/**
+ * Example using custom Flags with SecondBase.
```
Comment (neumayer) [inline example/src/main/java/org/secondbase/example/main/HelloSecondBase.java]: Use @link maybe?

File: core/src/main/java/org/secondbase/core/SecondBase.java (hunk 6 above)
Comment (acidmoose) [reply core/src/main/java/org/secondbase/core/SecondBase.java:40]: 0 is "no error code"

File: core/src/main/java/org/secondbase/core/ModuleConfigManager.java (hunk 2 above)
Comment (acidmoose) [reply core/src/main/java/org/secondbase/core/ModuleConfigManager.java]: done

File: core/src/main/java/org/secondbase/core/ModuleConfigManager.java (hunk 1 above)
Comment (acidmoose) [reply core/src/main/java/org/secondbase/core/ModuleConfigManager.java]: done

File: core/src/main/java/org/secondbase/core/ModuleConfigManager.java (hunk 3 above)
Comment (acidmoose) [reply core/src/main/java/org/secondbase/core/ModuleConfigManager.java]: done

File: core/src/main/java/org/secondbase/core/SecondBase.java (hunk 4 above)
Comment (acidmoose) [reply core/src/main/java/org/secondbase/core/SecondBase.java:14]: done

File: core/src/main/java/org/secondbase/core/SecondBase.java (hunk 5 above)
Comment (acidmoose) [reply core/src/main/java/org/secondbase/core/SecondBase.java]: done

File: core/src/main/java/org/secondbase/core/config/ConfigurableModule.java (hunk 7 above)
Comment (acidmoose) [reply core/src/main/java/org/secondbase/core/config/ConfigurableModule.java]: done

File: core/src/main/java/org/secondbase/core/config/ConfigurableModule.java (hunk 8 above)
Comment (acidmoose) [reply core/src/main/java/org/secondbase/core/config/ConfigurableModule.java]: done

File: example/src/main/java/org/secondbase/example/main/HelloSecondBase.java (hunk 9 above)
Comment (acidmoose) [reply example/src/main/java/org/secondbase/example/main/HelloSecondBase.java]: done

**Review by neumayer — APPROVED** [review]

**Pull request merged**